### PR TITLE
Modeling Users

### PR DESCRIPTION
**Pull request by brokenheart1712**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class User < ActiveRecord::Base
+  before_save { self.email = email.downcase }
+  validates :name,  presence: true, length: { maximum: 50 }
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
```
Comment (NguyenTanDuc) [inline app/models/user.rb]: Must be at top of file

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+  validates :email, presence: true, length: { maximum: 255 },
+                    format: { with: VALID_EMAIL_REGEX },
+                    uniqueness: { case_sensitive: false }
+  has_secure_password
```
Comment (NguyenTanDuc) [inline app/models/user.rb]: must be bottom of file

**Comment by NguyenTanDuc** [comment]
Checked. OK.

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class User < ActiveRecord::Base
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  before_save { self.email = email.downcase }
```
Comment (dieunb) [inline app/models/user.rb]: slice this line to anther block code. coding convention

**Pull request closed without merging**